backend/homework/9/src/main/java/com/example/demo/exceptions/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package com.example.demo.exceptions;
+
+
+import com.example.demo.dto.ErrorDTO;
+import com.example.demo.exceptions.custom.BadRequestException;
+import com.example.demo.exceptions.custom.ResourceNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+@ControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler(value = {ResourceNotFoundException.class})
+    public ResponseEntity<ErrorDTO> handleResourceNotFoundexception(ResourceNotFoundException example){
+        ErrorDTO error = new ErrorDTO(example.getMessage() + "Error in finding bool in library", HttpStatus.NOT_FOUND.value());
+        return new ResponseEntity<>(error, HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Since we are already catching an exception and know the cause, Use more precise response status instead of `INTERNAL_SERVER_ERROR` throughout.